File: montecarlo-pi/montecarlo-pi.f90
program fortran
   implicit none
   integer :: z
   integer :: N
   real :: pi
   real :: x
   real :: y
   integer :: i
   
   N = 10000000
   z = 0

   do i=1,N
      x = rand()
	  y = rand()
      if (x**2+y**2 < 1) then
         z = z+1
      end if
   end do

   pi = 4*REAL(z)/REAL(N)

   print *, 'Estimation de pi:  ',pi
end program fortran

File: montecarlo-pi/montecarlo-pi.c
#include <stdio.h>
#include <stdlib.h>

int main() {
	int i, N, z;
	double x, y, pi;
	
	N = 10000000;
	z = 0;

	for (i = 0; i < N; i++) {
		x = (double)rand() / (double)RAND_MAX;
		y = (double)rand() / (double)RAND_MAX;
		if( (x*x+y*y)< 1) z++; 
	}

	pi = 4*(double)z/(double)N ;

	printf("Estimation de pi: %f\n", pi);
	return 0;
}